application/jira/v0/issues.go

```diff
@@ -0,0 +1,318 @@
+}
+
+// https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-search/#api-rest-api-2-search-get
+// https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-search/#api-rest-api-2-search-post
+func (jiraClient *Client) nextGenIssuesSearch(_ context.Context, opt nextGenSearchRequest) (*resty.Response, error) {
+	var debug DebugSession
+	debug.SessionStart("nextGenIssuesSearch", StaticVerboseLevel)
+	defer debug.SessionEnd()
+
+	debug.AddMessage("opt:")
+	debug.AddRawMessage(opt)
+	var err error
+	apiEndpoint := "/rest/api/2/search"
+
+	req := jiraClient.Client.R().SetResult(&ListIssuesResp{})
+	var resp *resty.Response
+	if len(opt.JQL) < 50 {
+		// 50 is an arbitrary number to determine if the JQL is too long to be a query param
+		if err := addQueryOptions(req, opt); err != nil {
+			return nil, err
+		}
+		resp, err = req.Get(apiEndpoint)
+	} else {
+		req.SetBody(opt)
+		resp, err = req.Post(apiEndpoint)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	debug.AddMapMessage("Query", req.QueryParam)
+	debug.AddMessage("Status", resp.Status())
+	return resp, nil
+}
+
+var JQLReservedWords = []string{"a", "an", "abort", "access", "add", "after", "alias", "all", "alter", "and", "any", "are", "as", "asc", "at", "audit", "avg", "be", "before", "begin", "between", "boolean", "break", "but", "by", "byte", "catch", "cf", "char", "character", "check", "checkpoint", "collate", "collation", "column", "commit", "connect", "continue", "count", "create", "current", "date", "decimal", "declare", "decrement", "default", "defaults", "define", "delete", "delimiter", "desc", "difference", "distinct", "divide", "do", "double", "drop", "else", "empty", "encoding", "end", "equals", "escape", "exclusive", "exec", "execute", "exists", "explain", "false", "fetch", "file", "field", "first", "float", "for", "from", "function", "go", "goto", "grant", "greater", "group", "having", "identified", "if", "immediate", "in", "increment", "index", "initial", "inner", "inout", "input", "insert", "int", "integer", "intersect", "intersection", "into", "is", "isempty", "isnull", "it", "join", "last", "left", "less", "like", "limit", "lock", "long", "max", "min", "minus", "mode", "modify", "modulo", "more", "multiply", "next", "no", "noaudit", "not", "notin", "nowait", "null", "number", "object", "of", "on", "option", "or", "order", "outer", "output", "power", "previous", "prior", "privileges", "public", "raise", "raw", "remainder", "rename", "resource", "return", "returns", "revoke", "right", "row", "rowid", "rownum", "rows", "select", "session", "set", "share", "size", "sqrt", "start", "strict", "string", "subtract", "such", "sum", "synonym", "table", "that", "the", "their", "then", "there", "these", "they", "this", "to", "trans", "transaction", "trigger", "true", "uid", "union", "unique", "update", "user", "validate", "values", "view", "was", "when", "whenever", "where", "while", "will", "with"}
```

Comment:
question
What is this for?

Reply:
If we let the user input a JQL query directly in the future, maybe we can do some pre-processing to the string by utilizing this array.